### PR TITLE
Update `visibility` rule for stabilized `pub(restricted)` syntax.

### LINKED ISSUE #118: Can't parse pub(restricted) syntax
```rust
let raw = r#"
    pub(in m) struct Z(pub(in m::n) u8);
"#;

let actual = parse_item(raw).unwrap(); // Fails
```

The existing `visibility` rule needs to be updated to parse the `in` token.

Affected files:

- [`tests/rust/src/test/ui/resolve/auxiliary/privacy-struct-ctor.rs`](https://github.com/rust-lang/rust/blob/ddc5d7bd4b9ea3e8a8ccf82cb443e950be311d61/src/test/ui/resolve/auxiliary/privacy-struct-ctor.rs#L15)

### PR DESCRIPTION
**Pull request by Arnavion**
Fixes #118

### CODE REVIEW

File: src/data.rs [hunk 1]
```diff
@@ -287,6 +304,7 @@ mod printing {
                 Visibility::Restricted(ref path) => {
                     tokens.append("pub");
                     tokens.append("(");
+                    tokens.append("in");
```
Comment (Arnavion) [inline src/data.rs]: This converts `pub(self)` to `pub(in self)` (and same for `super`) which technically has the same semantics but also isn't exactly equal to the input.

Comment (dtolnay) [inline src/data.rs]: Please add a special case when printing that omits "in" if the path is self or super. This is what Rust does. You can see it by running cargo expand on the following code.

```rust
pub(self) fn f() {}
pub(in self) fn g() {}

fn main() {}
```

Comment (Arnavion) [inline src/data.rs]: Right, but that would also affect `test_round_trip`, just in the reverse direction. If the input contains `in super` the output would contain `super`. Though there isn't such a file right now, so I guess it's fine.

Comment (dtolnay) [inline src/data.rs]: That's fine right? Even if there were such a file, syntex would parse the before and after into the same AST.

Comment (Arnavion) [inline src/data.rs]: Oh right, the AST is compared, not the string. Yes, then it'll be fine.

**Pull request merged**